pkg/snapstore/backup_copier.go

```diff
@@ -0,0 +1,338 @@
+// SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package snapstore
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"sync"
+	"time"
+
+	brtypes "github.com/gardener/etcd-backup-restore/pkg/types"
+
+	"github.com/sirupsen/logrus"
+)
+
+// BackupCopier handles copying unique backups from primary to secondary snapstore
+type BackupCopier struct {
+	primary           brtypes.SnapStore
+	secondary         brtypes.SnapStore
+	logger            *logrus.Entry
+	config            *BackupCopierConfig
+	stopCh            chan struct{}
+	lastSyncTimestamp time.Time
+	mu                sync.RWMutex
+	running           bool
+}
+
+// BackupCopierConfig holds configuration for the backup copier
+type BackupCopierConfig struct {
+	// SyncPeriod defines how often to check for new backups to copy
+	SyncPeriod time.Duration
+	// MaxRetries defines maximum number of retries for failed copy operations
+	MaxRetries int
+	// RetryBackoff defines the backoff duration between retries
+	RetryBackoff time.Duration
+	// ConcurrentCopies defines maximum number of concurrent copy operations
+	ConcurrentCopies int
+}
+
+// DefaultBackupCopierConfig returns default configuration for backup copier
+func DefaultBackupCopierConfig() *BackupCopierConfig {
+	return &BackupCopierConfig{
+		SyncPeriod:       5 * time.Minute,
+		MaxRetries:       3,
+		RetryBackoff:     30 * time.Second,
+		ConcurrentCopies: 2,
+	}
+}
+
+// NewBackupCopier creates a new BackupCopier instance
+func NewBackupCopier(primary, secondary brtypes.SnapStore, config *BackupCopierConfig, logger *logrus.Entry) *BackupCopier {
+	if config == nil {
+		config = DefaultBackupCopierConfig()
```

Comment:
https://github.com/gardener/etcd-backup-restore/pull/917#discussion_r2498228416
```suggestion
		config = defaultBackupCopierConfig()
```